### PR TITLE
[TEST] Rename w3c_tracecontext_test to w3c_tracecontext_http_test_server

### PR DESCRIPTION
**Pull request by psx95**
This rename was requested in PR #3115 to make it more clear that the `test` artifact being built under `ext` for w3c trace context compliance is actually a test server.

See [this comment](https://github.com/open-telemetry/opentelemetry-cpp/pull/3115#discussion_r1831442311) requesting this change.

## Changes
 - Renames the `ext/w3c_tracecontext_test` directory to `ext/w3c_tracecontext_test_server`.
 - Updates the CI running W3C Trace Context Validation tests to accommodate this rename.

The CMake option has been kept as-is: `BUILD_W3CTRACECONTEXT_TEST`. 

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -946,8 +946,8 @@ jobs:
           CXX_STANDARD: '14'
         run: |
           ./ci/do_ci.sh cmake.w3c.trace-context.build-server
-          cd $HOME/build/ext/test/w3c_tracecontext_test
-          ./w3c_tracecontext_test &
+          cd $HOME/build/ext/test/w3c_tracecontext_test_server
+          ./w3c_tracecontext_test_server &
```
Comment (lalitb) [inline .github/workflows/ci.yml:950]: if we want to change it `w3c_tracecontext_http_test_server` would be better, to emphasize this is actually HTTP server. @ThomsonTan  - what you suggest ?

Comment (marcalff) [reply .github/workflows/ci.yml:950]: I agree with `w3c_tracecontext_http_test_server`, it makes it clearer that the way to talk to it is with curl.

**Review by marcalff — APPROVED** [review]
LGTM,

with renaming to `w3c_tracecontext_http_test_server` per @lalitb suggestion.

**Review by lalitb — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ThomsonTan) [reply .github/workflows/ci.yml:950]: +1.

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**